plutus-emulator/src/Wallet/Emulator/Types.hs

```diff
@@ -121,9 +114,13 @@ addBlocks :: Eff.Members EmulatorEffs effs => Integer -> Eff.Eff effs [Block]
 addBlocks i = traverse (const processPending) [1..i]
 
 -- | Add a number of blocks, notifying all the given 'Wallet's after each block.
-addBlocksAndNotify :: Eff.Members EmulatorEffs effs => [Wallet] -> Integer -> Eff.Eff effs ()
+addBlocksAndNotify :: Eff.Members EmulatorEffs effs => [Wallet] -> Integer -> Eff.Eff effs [Block]
 addBlocksAndNotify wallets i =
-    traverse_ (\_ -> processPending >>= walletsNotifyBlock wallets) [1..i]
+    let run _ = do
+            block <- processPending
+            walletsNotifyBlock wallets block
+            pure block
+    in traverse run [1..i]
```

Comment:
I find myself wondering if this is neater as:

```
    let run = do
            block <- processPending
            walletsNotifyBlock wallets block
            pure block
    in for [1..i] (const run)
```

Not sure it makes a difference. Still, I've typed it now. _press send_

Reply:
How about `replicateM` ?

Reply:
`replicateM` takes an `Int` :(

I'll just change it to `traverse (const run) [1..i]`